src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -90,36 +90,33 @@ template <typename Frame>
 tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
     const tnsr::ii<DataVector, 3, Frame>& grad_normal,
     const tnsr::i<DataVector, 3, Frame>& unit_normal_one_form,
-    const tnsr::II<DataVector, 3, Frame>& upper_spatial_metric) noexcept {
-  // Form projector
-  tnsr::Ij<DataVector, 3, Frame> projector(get<0>(unit_normal_one_form).size(),
-                                           0.0);
-  const auto& unit_normal_vector =
-      raise_or_lower_index(unit_normal_one_form, upper_spatial_metric);
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector) noexcept {
+  // Project symmetrized gradient
+  auto extrinsic_curvature(grad_normal);
+
+  Scalar<DataVector> nInJgradnij(get<0, 0>(grad_normal).size(), 0.0);
   for (size_t i = 0; i < 3; ++i) {
-    projector.get(i, i) += 1.0;
     for (size_t j = 0; j < 3; ++j) {
-      projector.get(i, j) -=
-          unit_normal_vector.get(i) * unit_normal_one_form.get(j);
+      nInJgradnij.get() += unit_normal_vector.get(i) *
+                           unit_normal_vector.get(j) * grad_normal.get(i, j);
     }
   }
 
-  // Project symmetrized gradient
-  tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
-      get<0>(unit_normal_one_form).size(), 0.0);
   for (size_t i = 0; i < 3; ++i) {
     for (size_t j = i; j < 3; ++j) {
+      extrinsic_curvature.get(i, j) += unit_normal_one_form.get(i) *
+                                       unit_normal_one_form.get(j) *
+                                       nInJgradnij.get();
       for (size_t k = 0; k < 3; ++k) {
-        for (size_t l = 0; l < 3; ++l) {
-          // Note: grad_normal is already symmetric, so
-          // no need to symmetrize it
-          extrinsic_curvature.get(i, j) +=
-              projector.get(k, i) * projector.get(l, j) * grad_normal.get(k, l);
-        }
+        extrinsic_curvature.get(i, j) -= unit_normal_vector.get(k) *
```

Comment:
Because we use expression templates to me it seems it would be better to write this as:

```
extrinsic_curvature.get(i, j) -= (unit_normal_vector.get(k) *
                                         unit_normal_one_form.get(i) *
                                         grad_normal.get(j, k) + unit_normal_vector.get(k) *
                                         unit_normal_one_form.get(j) *
                                         grad_normal.get(i, k));
```
(please ignore my terrible formatting)